temporalcli/commandsgen/docs.go

```diff
@@ -75,27 +79,75 @@ func (w *docWriter) writeCommand(c *Command) {
 }
 
 func (w *docWriter) writeSubcommand(c *Command) {
-	fileName := strings.Split(c.FullName, " ")[1]
-	subCommand := strings.Join(strings.Split(c.FullName, " ")[2:], "")
-	w.fileMap[fileName].WriteString("## " + subCommand + "\n\n")
+	fileName := c.fileName()
+	prefix := strings.Repeat("#", c.depth())
+	w.fileMap[fileName].WriteString(prefix + " " + c.leafName() + "\n\n")
 	w.fileMap[fileName].WriteString(c.Description + "\n\n")
-	w.fileMap[fileName].WriteString("Use the following options to change the behavior of this command.\n\n")
 
-	// gather options from command and all options aviailable from parent commands
-	var allOptions = make([]Option, 0)
-	for _, options := range w.optionsStack {
-		allOptions = append(allOptions, options...)
+	if w.isLeafCommand(c) {
+		w.fileMap[fileName].WriteString("Use the following options to change the behavior of this command.\n\n")
+
+		// gather options from command and all options aviailable from parent commands
+		var options = make([]Option, 0)
+		var globalOptions = make([]Option, 0)
+		for i, o := range w.optionsStack {
+			if i == len(w.optionsStack)-1 {
+				options = append(options, o...)
+			} else {
+				globalOptions = append(globalOptions, o...)
+			}
+		}
+
+		// alphabetize options
+		sort.Slice(options, func(i, j int) bool {
+			return options[i].Name < options[j].Name
+		})
+
+		sort.Slice(globalOptions, func(i, j int) bool {
+			return globalOptions[i].Name < globalOptions[j].Name
+		})
+
+		w.writeOptions("Flags", options, c)
+		w.writeOptions("Global Flags", globalOptions, c)
+
+	}
+}
+
+func (w *docWriter) writeOptions(prefix string, options []Option, c *Command) {
+	if len(options) == 0 {
+		return
 	}
 
-	// alphabetize options
-	sort.Slice(allOptions, func(i, j int) bool {
-		return allOptions[i].Name < allOptions[j].Name
-	})
+	fileName := c.fileName()
 
-	for _, option := range allOptions {
-		w.fileMap[fileName].WriteString(fmt.Sprintf("## %s\n\n", option.Name))
-		w.fileMap[fileName].WriteString(option.Description + "\n\n")
+	w.fileMap[fileName].WriteString(fmt.Sprintf("**%s:**\n\n", prefix))
 
+	for _, o := range options {
+		// option name and alias
+		w.fileMap[fileName].WriteString(fmt.Sprintf("**--%s** _%s_", o.Name, o.Type))
+		if len(o.Short) > 0 {
```

Comment:
Do we need to print the type for each form of the option?
i.e. for `**--output** _string-enum_, **-o** _string-enum_`

`**--output**, **-o** _string-enum_` would be more succinct. The type should always be the same, so I don't see why we need to print it twice

Reply:
Agree will update.